core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -620,8 +620,15 @@ object SparkSubmit {
 
     if (isKubernetesCluster) {
       childMainClass = "org.apache.spark.deploy.kubernetes.submit.Client"
-      childArgs += args.primaryResource
-      childArgs += args.mainClass
+      if (args.isPython) {
+        childArgs += args.primaryResource
```

Comment:
`val mainAppResource = args(0)`
so it is necessary to point to the mainAppResource in Client.main()